src/audio/aria/aria.c

```diff
@@ -133,7 +133,7 @@ static int init_aria(struct comp_dev *dev, struct comp_ipc_config *config,
 	list_init(&dev->bsource_list);
 	list_init(&dev->bsink_list);
 
-	dcache_invalidate_region(spec, sizeof(*aria));
+	dcache_invalidate_region((__force void __cache *)spec, sizeof(*aria));
```

Comment:
Fwiw, this does not parse easily - can we have a short hand macro for ```__force void __cache```

Reply:
please, let's not do that. I think that's the easiest we can do. There should be as few of these as possible. In fact this seems quite wrong to me. AFAICS `spec` comes from ipc3/helper.c or ipc4/helper.c as, e.g. (ipc3)
```
struct comp_dev *comp_new(struct sof_ipc_comp *comp)
{
	union ipc_config_specific spec;
	...
	comp_specific_builder(comp, &spec);
	cdev = drv->ops.create(drv, &config, &spec);
	...
}
```
So, `spec` is an object on stack, i.e. cached under Zephyr. `comp_specific_builder()` initialises it, so the above `dcache_invalidate_region()` potentially destroys that initialisation.

Reply:
...also, such type casts between different address spaces should only be done in very few places in system code, not in "user / application code." No application code should need to do that